test_add_static_path_resolution failure with /tmp and /private/tmp

### Describe the bug

On macOS, test_add_static_path_resolution fails like this:
```
______________________________________________________________________________________________ test_add_static_path_resolution _______________________________________________________________________________________________
[gw0] darwin -- Python 3.8.13 /sw/bin/python3.8

router = <aiohttp.web_urldispatcher.UrlDispatcher object at 0x109647880>

    def test_add_static_path_resolution(router: any) -> None:
        """Test that static paths are expanded and absolute."""
        res = router.add_static("/", "~/..")
        directory = str(res.get_info()["directory"])
>       assert directory == str(pathlib.Path.home().parent)
E       AssertionError: assert '/private/tmp' == '/tmp'
E         - /tmp
E         + /private/tmp

directory  = '/private/tmp'
res        = <StaticResource   -> PosixPath('/private/tmp')>
router     = <aiohttp.web_urldispatcher.UrlDispatcher object at 0x109647880>

tests/test_urldispatch.py:354: AssertionError
```

On macOS, /tmp is a symlink to /private/tmp:
```
nieder $ ls -ld /tmp /private/tmp
drwxrwxrwt  535 root  wheel  17120 Feb  9 06:58 /private/tmp/
lrwxr-xr-x@   1 root  wheel     11 May  4  2021 /tmp@ -> private/tmp
```

### To Reproduce

1) Build aiohttp, then run tests in source dir:
```
/sw/bin/python3.8 -m pytest -vv || exit 2
/sw/lib/python3.8/site-packages/pytest_benchmark/logger.py:46: PytestBenchmarkWarning: Benchmarks are automatically disabled because xdist plugin is active.Benchmarks cannot be performed reliably in a parallelized environment.
  warner(PytestBenchmarkWarning(text))
==================================================================================================== test session starts =====================================================================================================
platform darwin -- Python 3.8.13, pytest-7.4.4, pluggy-1.4.0 -- /sw/bin/python3.8
codspeed: 2.2.1 (callgraph: not supported)
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/sw/build.build/aiohttp-py38-3.10.11-1/aiohttp-3.10.11/.hypothesis/examples')
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
Using --randomly-seed=726549581
rootdir: /sw/build.build/aiohttp-py38-3.10.11-1/aiohttp-3.10.11
configfile: setup.cfg
testpaths: tests/
plugins: cov-6.0.0, codspeed-2.2.1, hypothesis-6.42.1, flaky-3.8.1, timeout-2.3.1, benchmark-3.4.1, randomly-3.15.0, asyncio-0.21.1, datadir-1.5.0, pyfakefs-5.3.4, xdist-3.5.0, mock-3.12.0
asyncio: mode=strict
4 workers [3068 items] error / 1 skipped                                                                                                                                                                                     
scheduling tests via LoadScheduling
```
2) error output as seen above

### Expected behavior

The test should pass.

### Logs/tracebacks

```python-traceback
See above output for error report from this test.
```

### Python Version

```console
$ python --version
3.8.13

but also happens with python3.10.4
```

### aiohttp Version

```console
This is trying to build aiohttp v3.10.11
```

### multidict Version

```console
Name: multidict
Version: 6.1.0
Summary: multidict implementation
Home-page: https://github.com/aio-libs/multidict
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /sw/lib/python3.8/site-packages
Requires: typing-extensions
Required-by: yarl
```

### propcache Version

```console
Name: propcache
Version: 0.2.0
Summary: Accelerated property cache
Home-page: https://github.com/aio-libs/propcache
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache-2.0
Location: /sw/lib/python3.8/site-packages
Requires: 
Required-by: yarl
```

### yarl Version

```console
Name: yarl
Version: 1.15.2
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache-2.0
Location: /sw/lib/python3.8/site-packages
Requires: idna, multidict, propcache
Required-by:
```

### OS

macOS 10.14.6

### Related component

Server, Client

### Additional context

_No response_

### Code of Conduct

- [x] I agree to follow the aio-libs Code of Conduct

--- a/tests/test_urldispatch.py
+++ b/tests/test_urldispatch.py
@@ -373,7 +373,7 @@ def test_add_static_path_resolution(router: web.UrlDispatcher) -> None:
     """Test that static paths are expanded and absolute."""
     res = router.add_static("/", "~/..")
     directory = str(res.get_info()["directory"])
-    assert directory == str(pathlib.Path.home().parent)
+    assert directory == str(pathlib.Path.home().resolve(strict=True).parent)
 
 
 def test_add_static(router: web.UrlDispatcher) -> None:

